client/src/stores/collectionElementsStore.ts

```diff
@@ -1,32 +1,45 @@
 import { defineStore } from "pinia";
 import Vue, { computed, ref } from "vue";
 
-import { DCESummary, HDCASummary, HistoryContentItemBase } from "./services";
+import { CollectionEntry, DCESummary, HDCASummary, HistoryContentItemBase, isHDCA } from "./services";
 import * as Service from "./services/datasetCollection.service";
 
 export const useCollectionElementsStore = defineStore("collectionElementsStore", () => {
     const storedCollections = ref<{ [key: string]: HDCASummary }>({});
     const loadingCollectionElements = ref<{ [key: string]: boolean }>({});
     const storedCollectionElements = ref<{ [key: string]: DCESummary[] }>({});
 
+    /**
+     * Returns a key that can be used to store or retrieve the elements of a collection in the store.
+     *
+     * It consistently returns a DatasetCollection ID for (top level) HDCAs or sub-collections.
+     */
+    function getCollectionKey(collection: CollectionEntry): string {
```

Comment:
We should always know what we're dealing with so we don't need to make runtime guesses here.

Reply:
Although it is pretty explicit ... it'll work for now.